Program.cs

```diff
@@ -1,143 +1,83 @@
 
 public class Program
 {
+    private static readonly HttpClient HttpClient = new HttpClient();
+
     public static async Task Main(string[] args)
     {
         // Your API token is already set in this variable
-        string apiToken = Environment.GetEnvironmentVariable("ASSEMBLY_AI_API_KEY"); // only pre-modification for git
+        var apiToken = Environment.GetEnvironmentVariable("ASSEMBLY_AI_API_KEY"); // only pre-modification for git
+        HttpClient.DefaultRequestHeaders.Authorization = new AuthenticationHeaderValue(apiToken);
 
         // -----------------------------------------------------------------------------
         // Update the file path here, pointing to a local audio or video file.
         // If you don't have one, download a sample file: https://storage.googleapis.com/aai-web-samples/espn-bears.m4a
         // You may also remove the upload step and update the 'audio_url' parameter in the
         // 'GetTranscriptAsync' function to point to a remote audio or video file.
         // -----------------------------------------------------------------------------
-        var path = "./espn-bears.m4a";
-        var fileUploader = new FileUploader();
-        var uploadUrl = await fileUploader.UploadFileAsync(apiToken, path);
+        var path = "C:\\Users\\niels\\Downloads\\espn-bears.m4a";
+        var uploadedFileUrl = await UploadFileAsync(path);
+        var transcript = await GetTranscriptAsync(uploadedFileUrl);
+        Console.WriteLine("Transcript:\n" + transcript);
+    }
 
-        TranscriptFetcher transcriptFetcher = new TranscriptFetcher();
+    private static async Task<string> UploadFileAsync(string path)
+    {
+        await using var fileStream = File.OpenRead(path);
+        using var fileContent = new StreamContent(fileStream);
+        fileContent.Headers.ContentType = new MediaTypeHeaderValue("application/octet-stream");
 
-        try
-        {
-            dynamic transcript = await transcriptFetcher.GetTranscriptAsync(apiToken, uploadUrl);
+        using var response = await HttpClient.PostAsync("https://api.assemblyai.com/v2/upload", fileContent);
 
-            Console.WriteLine("Transcript:\n" + transcript.text);
-        }
-        catch (Exception ex)
+        if (response.IsSuccessStatusCode == false)
+            throw new Exception($"Error: {response.StatusCode} - {response.ReasonPhrase}");
+
+        var jsonDoc = await response.Content.ReadFromJsonAsync<JsonDocument>();
+        return jsonDoc.RootElement.GetProperty("upload_url").GetString();
+    }
+
+    private static async Task<string> GetTranscriptAsync(string audioUrl)
+    {
+        var data = new {audio_url = audioUrl};
+        var content = new StringContent(JsonSerializer.Serialize(data), Encoding.UTF8, "application/json");
+
+        using var response = await HttpClient.PostAsync("https://api.assemblyai.com/v2/transcript", content);
+        var responseJson = await response.Content.ReadFromJsonAsync<JsonDocument>();
```

Comment:
You could replace `JsonDocument` with typed classes so you don't have to manually parse the JSON, but since it's simple enough I stuck to `JsonDocument`.